### PR TITLE
gh-105539: Fix ResourceWarning from unclosed SQLite connection in ClosedCurTests.

### LINKED ISSUE #105539: Emit resource warning if sqlite3 fails to close the database
_See https://github.com/python/cpython/issues/103837#issuecomment-1541808307 and https://github.com/python/cpython/issues/103837#issuecomment-1543936079_:

EAA:
> I still think we should consider changing the underlying API so we can emit a resource warning if close failed.

VS:
> IMO sqlite3 should be modified to emit a ResourceWarning.
>
> Currently, no ResourceWarning is emitted when a connection is not closed explicitly:

### Linked PRs
* gh-108015
* gh-108017
* gh-108360
* gh-108565

### PR DESCRIPTION
**Pull request by felixxm**
Follow up to 1a1bfc28912a39b500c578e9f10a8a222638d411.

* Issue: gh-105539

### CODE REVIEW

File: Lib/test/test_sqlite3/test_dbapi.py
Comment (erlend-aasland) [inline Lib/test/test_sqlite3/test_dbapi.py:1]: Here's a slightly smaller diff:

```diff
diff --git a/Lib/test/test_sqlite3/test_dbapi.py b/Lib/test/test_sqlite3/test_dbapi.py
index d80ad7af32..f6f4079484 100644
--- a/Lib/test/test_sqlite3/test_dbapi.py
+++ b/Lib/test/test_sqlite3/test_dbapi.py
@@ -40,6 +40,7 @@
 from test.support.os_helper import TESTFN, TESTFN_UNDECODABLE, unlink, temp_dir, FakePath
 
 from .util import memory_database, cx_limit
+from .util import MemoryDatabaseMixin
 
 
 class ModuleTests(unittest.TestCase):
@@ -1740,10 +1741,9 @@ def test_closed_call(self):
         self.check(self.con)
 
 
-class ClosedCurTests(unittest.TestCase):
+class ClosedCurTests(MemoryDatabaseMixin, unittest.TestCase):
     def test_closed(self):
-        con = sqlite.connect(":memory:")
-        cur = con.cursor()
+        cur = self.con.cursor()
         cur.close()
 
         for method_name in ("execute", "executemany", "executescript", "fetchall", "fetchmany", "fetchone"):
```

But there are more resource leaks; we need to continue hunting ;)

Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_dbapi.py:1]: `test_backup` leaks!

Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_dbapi.py:1]:
```diff
diff --git a/Lib/test/test_sqlite3/test_backup.py b/Lib/test/test_sqlite3/test_backup.py
index 4584d976bc..c7400d8b21 100644
--- a/Lib/test/test_sqlite3/test_backup.py
+++ b/Lib/test/test_sqlite3/test_backup.py
@@ -137,7 +137,7 @@ def progress(status, remaining, total):
             raise SystemError('nearly out of space')
 
         with self.assertRaises(SystemError) as err:
-            with sqlite.connect(':memory:') as bck:
+            with memory_database() as bck:
                 self.cx.backup(bck, progress=progress)
         self.assertEqual(str(err.exception), 'nearly out of space')
 

```

Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_dbapi.py:1]: All right, that's it; I get no more leaks now :)

**Review by erlend-aasland — APPROVED** [review]

File: Lib/test/test_sqlite3/test_dbapi.py
Comment (felixxm) [reply Lib/test/test_sqlite3/test_dbapi.py:1]: Thanks! I didn't know we have such mixins.

Comment (erlend-aasland) [reply Lib/test/test_sqlite3/test_dbapi.py:1]: No worries, they were committed recently :) Thanks for the PR!

**Comment by felixxm** [comment]
@erlend-aasland I found and fixed one more:
```
./python -m test -R : test_audit
```

**Comment by erlend-aasland** [comment]
> @erlend-aasland I found and fixed one more:
> 
> ```
> ./python -m test -R : test_audit
> ```

Good catch!

**Comment by erlend-aasland** [comment]
Thanks, Mariusz!

**Pull request merged**